Epic6.5 sprint2 mk

mattkrick opened the pull request:
Meeting > Check-in
- [ ] create 2 teams. make sure the teamId starts with a different character. run through a meeting with each. confirm that they have different check-in questions. (#964)
- [ ] As the facilitator, click your own avatar. see that you can navigate to your section
- [ ] As the facilitator, click another user's avatar. see that you can navigate to their section or promote them to facilitator _if they are connected_
- [ ] In another window, as a non-facilitator, click the facilitator's avatar. see that you can request to become facilitator.
- [ ] When another user requests to become facilitator, the facilitator gets a toast with an option to promote
- [ ] if the facilitator clicks promote, the requestor becomes the facilitator
- [ ] all users can see their connection & checked in status by clicking the avatars

Meeting > Lobby
- [ ] create 1 agenda item, then delete it. (#1197)

My Dashboard
- [ ] no teams exist in the sidebar that shouldn't (Taya & maybe Jordan? can test by copying prod DB to local or staging) (#1058)

Settings > Organizations > Billing
- [ ] click upcoming invoice, see the upcoming invoice
- [ ] Extend the trial (see below), click upcoming invoice, see the upcoming invoice

Stripe
- [ ] Create an account, then extend the trial per docs in guru, get orgId from the Settings > Organizations > Organization url. Go to stripe, search by orgId. ensure that there is 1 ended subscription & 1 active one

mattkrick commented:
@jordanh this is ready for some :eyes:
the final 2 bugfixes are pretty large, so i'm gonna break em into their own.

mattkrick commented:
@jordanh this one needs your eyes, too.

jordanh commented on src/client/Atmosphere.js:
```diff
@@ -149,7 +149,7 @@ export default class Atmosphere extends Environment {
 
   ensureSubscription = (config) => {
-    const {subscription, variables} = config;
+    const {subscription, variables = {}} = config;
```
ah! good habit here!

jordanh commented on src/server/graphql/types/UserMemoType.js:
```diff
@@ -0,0 +1,12 @@
+import {FACILITATOR_REQUEST} from 'universal/subscriptions/constants';
+
+const UserMemoType = new GraphQLEnumType({
+  name: 'UserMemoType',
```
You gots ideas for other memos?

mattkrick replied:
yeah, I may change the name, but i got BIG plans for this.
basically, we're gonna have each user listening to a single USER_MEMO.userId channel.
Then, we can pass in anything we want to them, notifications, toasts, new auth tokens, etc.
i'm still weighing out the pros/cons of doing this for PRESENCE. at the expense of 1 DB hit per login, we could reduce the # of presence messages we send to the whole team.

jordanh commented on src/universal/utils/makeCheckinGreeting.js:
```diff
@@ -77,10 +77,14 @@ const questions = [
 
-export const makeCheckinGreeting = (week) => {
-  return greetings[week % greetings.length];
+export const makeCheckinGreeting = (week, seedId = '') => {
```
:)

jordanh commented:
# Test Cycle 1
## Status: In-Progress
## Tests
Meeting > Check-in
- [ ] create 2 teams. make sure the teamId starts with a different character. run through a meeting with each. confirm that they have different check-in questions. (#964)
- [ ] As the facilitator, click your own avatar. see that you can navigate to your section
- [ ] As the facilitator, click another user's avatar. see that you can navigate to their section or promote them to facilitator _if they are connected_
- [ ] In another window, as a non-facilitator, click the facilitator's avatar. see that you can request to become facilitator.
- [ ] When another user requests to become facilitator, the facilitator gets a toast with an option to promote
- [ ] if the facilitator clicks promote, the requestor becomes the facilitator
- [ ] all users can see their connection & checked in status by clicking the avatars

Meeting > Lobby
- [ ] create 1 agenda item, then delete it. (#1197)

My Dashboard
- [ ] no teams exist in the sidebar that shouldn't (Taya & maybe Jordan? can test by copying prod DB to local or staging) (#1058)

Settings > Organizations > Billing
- [ ] click upcoming invoice, see the upcoming invoice
- [ ] Extend the trial (see below), click upcoming invoice, see the upcoming invoice

Stripe
- [ ] Create an account, then extend the trial per docs in guru, get orgId from the Settings > Organizations > Organization url. Go to stripe, search by orgId. ensure that there is 1 ended subscription & 1 active one

jordanh commented:
I was getting that annoying `cannot find analytics.js` business again. I nuked my `yarn.lock` and `node_modules` and it went away. Perhaps nuke your `node_modules` and use the `yarn.lock` I am about to check in?

jordanh commented:
Ok @mattkrick, back over to you!

mattkrick commented:
fixed issues:
- [ ] when someone gets promoted to the facilitator, everyone who was in sync with the last facilitator is now in sync with the new facilitator. this was a longstanding undocumented bug on disconnects.
- [ ] graphiql now works

jordanh commented:
# Test Cycle 2
## Status: Complete
## Tests
Meeting > Check-in
- [ ] create 2 teams. make sure the teamId starts with a different character. run through a meeting with each. confirm that they have different check-in questions. (#964)
- [ ] As the facilitator, click your own avatar. see that you can navigate to your section
- [ ] As the facilitator, click another user's avatar. see that you can navigate to their section or promote them to facilitator _if they are connected_
- [ ] In another window, as a non-facilitator, click the facilitator's avatar. see that you can request to become facilitator.
- [ ] When another user requests to become facilitator, the facilitator gets a toast with an option to promote
- [ ] if the facilitator clicks promote, the requestor becomes the facilitator
- [ ] all users can see their connection & checked in status by clicking the avatars

Meeting > Lobby
- [ ] create 1 agenda item, then delete it. (#1197)


Settings > Organizations > Billing
- [ ] click upcoming invoice, see the upcoming invoice
- [ ] Extend the trial (see below), click upcoming invoice, see the upcoming invoice

Stripe
- [ ] Create an account, then extend the trial per docs in guru, get orgId from the Settings > Organizations > Organization url. Go to stripe, search by orgId. ensure that there is 1 ended subscription & 1 active one

- - -

## Results

## Not tested

My Dashboard
- [ ] no teams exist in the sidebar that shouldn't (Taya & maybe Jordan? can test by copying prod DB to local or staging) (#1058)

mattkrick commented:
A dry run results in the following output:
```
[migrate-rethinkdb]  ↑  up  ↑  20170824163600 removeBadTeamMembers
Removed 1 users from 5 teams
affected users:  Taya,Taya,Taya,Taya,Taya
```

jordanh commented:
I am getting this malarchy again:

![image](https://user-images.githubusercontent.com/103000/30083375-0c46146c-9243-11e7-9f4e-764f47a99510.png)

I've done the usual business (remove node_modules, yarn, remove yarn.lock, yarn). Wasn't this a router thing? Did we fix it on this branch? I forget...

mattkrick commented:
ah, that was #1324, already merged to master.
did you also nuke build & dll?
just did a fresh build & i am not seeing this. if the dll was referencing another version of react-router that might trigger it.

jordanh commented:
Let me try nuking build...

jordanh commented:
Gosh, I'm still getting that thing. Let me try master...

mattkrick commented:
```
git clone git@github.com:ParabolInc/action.git action-temp
cd action-temp
git checkout epic6.5-sprint2-mk
yarn && npm run build:relay && dev
```
if that doesn't work, then we got a big ol problem!

jordanh commented:
I turns out I am a big silly, silly pants. I was only doing `npm build` and not `npm build:dll`. Argh!

mattkrick commented:
oh phew! hey as long as it works i'm happy!

jordanh commented:
@mattkrick tests are complete, see: https://github.com/ParabolInc/action/pull/1296#issuecomment-327248721

We're close to merging. Probably just one more micro-pass.

mattkrick commented:
@jordanh that link goes to the comment that shows all boxes are checked (except for taya's bad teams on sidebar). anything left to do?

jordanh commented:
Have a look down at the Results section, I found one little bug-a-boo. It's a bit funny – all the tests as we labelled them pass, but there is still this one little thing :)

mattkrick commented:
ah! needed a refresh. github doesn't do evented subs all that well i guess :wink:.

of note, we may want to put another asterisk by the metrics last week, the `createFirstTeam` event probably never got sent, just fixed it.

also fixed the infinite loop problem. nice find!

jordanh commented:
Ok, going to do one more test cycle now...

jordanh commented:
# Test Cycle 3
## Status: In Progress
## Tests
Meeting > Check-in
- [ ] create 2 teams. make sure the teamId starts with a different character. run through a meeting with each. confirm that they have different check-in questions. (#964)
- [ ] As the facilitator, click your own avatar. see that you can navigate to your section
- [ ] As the facilitator, click another user's avatar. see that you can navigate to their section or promote them to facilitator _if they are connected_
- [ ] In another window, as a non-facilitator, click the facilitator's avatar. see that you can request to become facilitator.
- [ ] When another user requests to become facilitator, the facilitator gets a toast with an option to promote
- [ ] if the facilitator clicks promote, the requestor becomes the facilitator
- [ ] all users can see their connection & checked in status by clicking the avatars

- - -

## Results

jordanh commented:
LGTM!
